addons/mail/models/mail_alias.py

```diff
@@ -197,21 +199,49 @@ def get_alias_values(self):
         """
         return {'alias_parent_thread_id': self.id}
 
+    def _filter_alias_fields(self, vals):
+        """ Split the vals dict into two dictionnary of vals,
+            one for alias field and the other for other fields """
+        alias_field, other_field = {}, {}
+        for key in list(vals):
+            if key in self.ALIAS_WRITEABLE_FIELDS:
+                alias_field[key] = vals.get(key)
+            else:
+                other_field[key] = vals.get(key)
+        return alias_field, other_field
+
     @api.model
     def create(self, vals):
-        """ Create a record with ``vals``, and create a corresponding alias. """
-        record = super(AliasMixin, self.with_context(
-            alias_model_name=self.get_alias_model_name(vals),
-            alias_parent_model_name=self._name,
-        )).create(vals)
+        """ - Slip writable fields of mail.alias and other fields,
+            - Create a corresponding alias if none specify. 
+            - Create the record with the rigth alias and other fields value
+            - Update the alias with values of get_alias_values() and return the record """
+        alias_field, other_field = self._filter_alias_fields(vals)
+        if not other_field.get('alias_id', False):
+            alias = self.env['mail.alias'].sudo().with_context(
+                alias_model_name=self.get_alias_model_name(vals),
+                alias_parent_model_name=self._name,
+            ).create(alias_field)
+            other_field['alias_id'] = alias.id
+        record = super(AliasMixin, self).create(other_field)
         record.alias_id.sudo().write(record.get_alias_values())
         return record
 
+    def write(self, vals):
+        """ Split writable fields of mail.alias and other fields
+            alias fields will write with sudo and the other normally """
+        alias_field, other_field = self._filter_alias_fields(vals)
+        if alias_field and self.check_access_rights('write', raise_exception=False):
```

Comment:
I think we could call check_access_rights only if there is nothing in other_field, as that call to super should trigger the check. To avoid doing multiple checks we could simply do it manually when only alias fields are updated.